src/ipfsd-daemon.js

```diff
@@ -128,24 +130,32 @@ class Daemon {
       this.path = initOpts.directory
     }
 
-    const args = ['init', '-b', initOpts.keysize || 2048]
+    const bits = initOpts.bits ? initOpts.bits : this.bits
+    const args = ['init']
+    // do not just set a default keysize,
+    // in case we decide to change it at
+    // the daemon level in the future
+    if (bits) {
+      args.concat(['-b', bits])
+    }
     if (initOpts.pass) {
       args.push('--pass')
       args.push('"' + initOpts.pass + '"')
     }
+    log(`initializing with keysize: ${bits}`)
     run(this, args, { env: this.env }, (err, result) => {
       if (err) {
         return callback(err)
       }
 
-      configureNode(this, this.opts.config, (err) => {
-        if (err) {
-          return callback(err)
-        }
-
+      waterfall([
+        (cb) => this.getConfig(cb),
+        (conf, cb) => this.replaceConfig(defaults({}, this.opts.config, conf), cb)
+      ], (err) => {
+        if (err) { return callback }
         this.clean = false
         this.initialized = true
-        callback(null, this)
+        return callback(null, this)
```

Comment:
Init should not need to return a reference to the instance.

Reply:
this is done for backwards compatibility, that is how we used to do it previous versions, plus it helps with chaining in `async/waterfall` type constructs - but not a real contention point for me, if noone objects, I'll remove.

Reply:
please remove.